cupy/_core/core.pyx

```diff
@@ -620,7 +620,7 @@ cdef class _ndarray_base:
             newarray.data.copy_from_device_async(x.data, x.nbytes)
         return newarray
 
-    cpdef _ndarray_base view(self, dtype=None):
+    cpdef _ndarray_base view(self, dtype=None, array_class=None):
```

Comment:
This function should accept all of these:

```
a.view()
a.view(some_dtype)
a.view(dtype=some_dtype)
a.view(ndarray_subclass)
a.view(type=ndarray_subclass)  # <---
```

https://numpy.org/doc/stable/reference/generated/numpy.ndarray.view.html#numpy.ndarray.view

I think we will need `**kwargs` here...

Reply:
Thanks! I've added another `def view` method around the original `cdef`d one, as it seems that Cython does not take `*args` and `**kwargs` for `cdef`d methods.

.pyx codes that called the original `cdef view` can be also fixed to call `cdef _view_impl` instead of the new `def view`, but I'll make this change in another PR as we have tens of such lines and the next release is approaching.

```
$ find . -name '*.pyx' | xargs grep 'view(' | wc -l
50
```

Reply:
I found we can provide a wrapper of `view` method in `ndarray` class, which makes things easier. We can still use `cpdef view` in Cython files with `_ndarray_base` objects. [0b1343f](https://github.com/cupy/cupy/pull/6720/commits/0b1343f31460b7806b93af10bf625f5ec24da88e)